test/instances/router.lua

```diff
@@ -28,4 +31,21 @@ end
 box.cfg(helpers.box_cfg())
 box.schema.user.grant('guest', 'super', nil, nil, {if_not_exists = true})
 
+local function router_cfg(cfg, router)
+    local vshard_cfg = vcfg.extract_vshard(cfg)
+    local box_cfg = vcfg.extract_box(cfg, {})
+    box.cfg(box_cfg)
+    if router == nil then
+        -- Static router configuration
+        return vshard.router.cfg(vshard_cfg)
+    end
+    if type(router) == 'table' then
+        -- Named router reconfiguration
+        return router:cfg(vshard_cfg)
+    end
+    -- Named router creation
+    return vshard.router.new(router, vshard_cfg)
+end
+
+_G.router_cfg = router_cfg
```

Comment:
Have you checked if this is really needed in luatests? If so then why? Which of them are trying to give the router `box.cfg` options? I would suggest we try to leave this file intact and try to fix the luatests so as they would call `box.cfg` explicitly for whatever they need. If vtest is doing that, then we should fix vtest.

The reasoning is that I want this instance file to be doing as few hacks as possible. This would help to catch potential issues which could be hidden by our hacks otherwise.

Reply:
It's `replication_timeout`, which is set in `vtest.config_new`. Let's not set it there. Fixed